generator/src/main/java/com/scottlogic/deg/generator/guice/OutputTargetProvider.java

```diff
@@ -0,0 +1,34 @@
+package com.scottlogic.deg.generator.guice;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.scottlogic.deg.generator.generation.GenerationConfigSource;
+import com.scottlogic.deg.generator.outputs.targets.FileOutputTarget;
+import com.scottlogic.deg.generator.outputs.targets.OutputTarget;
+import com.scottlogic.deg.generator.outputs.targets.StreamOutputTarget;
+
+public class OutputTargetProvider implements Provider<OutputTarget> {
+
+    private final GenerationConfigSource configSource;
+    private final FileOutputTarget fileOutputTarget;
+    private final StreamOutputTarget streamOutputTarget;
+
+    @Inject
+    public OutputTargetProvider(GenerationConfigSource configSource, FileOutputTarget fileOutputTarget,
+                                StreamOutputTarget streamOutputTarget) {
+        this.configSource = configSource;
+        this.fileOutputTarget = fileOutputTarget;
+        this.streamOutputTarget = streamOutputTarget;
+    }
+
+    @Override
+    public OutputTarget get() {
+        switch (configSource.getOutputDestination()) {
```

Comment:
As before: default case to switch, more specific exception.

Given both classes seem to be doing the same thing this logic could be extracted to an abstract class, or the interface.